OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs

```diff
@@ -32,8 +32,17 @@ public override Activity Tick(Actor self)
 			if (host == null)
 				return NextActivity;
 
+			if (aircraft.IsPlane)
+				return ActivityUtils.SequenceActivities(
+					aircraft.GetResupplyActivities(host)
+					.Append(new CallFunc(() => aircraft.UnReserve()))
+					.Append(new WaitFor(() => Reservable.IsReserved(host)))
+					.Append(new TakeOff(self))
+					.Append(NextActivity).ToArray());
```

Comment:
SequenceActivities accepts a list as parameters, so building your own array shouldn't be needed.